r/src/compute.cpp

```diff
@@ -621,6 +621,14 @@ std::vector<std::string> compute__GetFunctionNames() {
   return arrow::compute::GetFunctionRegistry()->GetFunctionNames();
 }
 
+// [[arrow::export]]
+void compute__Initialize() {
+#if ARROW_VERSION_MAJOR >= 21
+  auto status = arrow::compute::Initialize();
+  StopIfNotOk(status);
+#endif
```

Comment:
Do we need the version guard here at all? We no longer support backwards compatibility with old libarrows in the R library (#43623), so I don't _think_ this is strictly necessary but I'm fine to keep it if it's helpful for some other reason?

Reply:
Thanks @jonkeane you are correct, I did miss your PR being merged, the check doesn't seem necessary now. I've pushed https://github.com/apache/arrow/pull/46261/commits/7526369c2c20db6c46c6b6943bb60d51f6e7eab0 removing the `#if ARROW_VERSION_MAJOR >= 21`